### ISSUE TITLE
kimi-k2: gate DeepEP decode graph-replay numerics (full-bucket graph-vs-eager)

### ISSUE THREAD

**Issue by xiaguan** [pegainfer-project/pegainfer#300]
Part of #221 (roadmap). Follow-up from #298 (DeepEP migration) / #227 (graph capture).

## Problem

DeepEP decode now captures into CUDA graphs and replays only at full bucket occupancy (`active_len == decode_batch_size`, always 8 under the DP scheduler) — `pegainfer-kimi-k2/src/runner/worker/state.rs:261`. The replay path has no dedicated numerics gate:

- `pegainfer-kimi-k2/tests/vllm_golden_gate.rs:300` runs with `enable_cuda_graph: false`.
- Even with graphs on, the gate's concurrent pass peaks at ~2 active rows/rank — it never fills the bucket, so it would only exercise the eager arm.

Current evidence is indirect: in the 8×H200 bs64 bench, 56/64 traces were byte-identical to the eager runs' dominant trace families. Good signal, but not a gate a fresh clone can re-run.

## Repro

```
cargo run --release --features kimi-k2 -- --model-path /data/models/Kimi-K2.6 --tp-size 1 --dp-size 8 --cuda-graph true
# drive >= 64 concurrent requests so every rank holds 8 active rows -> replay arm
```

## Acceptance

- A committed test (or gate mode) that drives all 8 rows/rank to full bucket with graphs enabled and compares decode output (token ids or logits) graph-vs-eager on the same prompts.
- Red if replay diverges from eager beyond the bf16 ULP floor.

## Related

- #228 — combine_impl tuning (91µs vs ~37µs NVLink theory), the remaining decode perf lever.
- Prefill alloc/spin cleanup (+2ms TTFT) is prefill-side, tracked in `docs/models/kimi-k2/deepep-migration.md`.
